packages/nextjs/hooks/scaffold-eth/useDeployedContractInfo.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from "react";
+import { useProvider } from "wagmi";
+import { toast } from "~~/utils/scaffold-eth";
+
+type GeneratedContractType = {
+  address: string;
+  abi: any[];
+};
+
+/**
+ * use this hook to get deployed contract from generated files of `yarn deploy`
+ * @param chainId - deployed contract chainId
+ * @param contractName - name of deployed contract
+ * @returns {GeneratedContractType | undefined} object containing contract address and abi or undefined if contract is not found
+ */
+export const useDeployedContractInfo = ({
+  chainId,
+  contractName,
+}: {
+  chainId: string | undefined;
+  contractName: string | undefined | null;
+}) => {
+  const [deployedContractData, setDeployedContractData] = useState<undefined | GeneratedContractType>(undefined);
+  const provider = useProvider();
+
+  useEffect(() => {
+    (async () => {
+      let ContractData;
+      try {
+        ContractData = require("~~/generated/hardhat_contracts.json");
+        const contractsAtChain = ContractData[chainId as keyof typeof ContractData];
+        const contractsData = contractsAtChain?.[0]?.contracts;
+
+        // Looking at blockchain to see whats stored at `deployedContractData.address`, if its `0x0` then high possibility that its not a contract
+        const code = await provider.getCode(contractsData.address);
```

Comment:
This wasn't getting the right address. I fixed it on https://github.com/scaffold-eth/se-2/pull/130/commits/4c93cb75162bb9529ada46d8f2bbb554e0ba7fb8

With some other minor tweaks